include/exadg/incompressible_navier_stokes/postprocessor/postprocessor.cpp

```diff
@@ -101,53 +104,267 @@ PostProcessor<dim, Number>::setup(Operator const & pde_operator)
 
 template<int dim, typename Number>
 void
-PostProcessor<dim, Number>::do_postprocessing(VectorType const & velocity,
-                                              VectorType const & pressure,
-                                              double const       time,
-                                              int const          time_step_number)
+PostProcessor<dim, Number>::do_postprocessing(VectorType const &     velocity,
+                                              VectorType const &     pressure,
+                                              double const           time,
+                                              types::time_step const time_step_number)
 {
+  // TODO: NOTE THAT CALLING THE FUNCTION HERE RESULTS IN SLOWER CODE. THIS IS REPORTED AN RESOLVED
+  // IN PR #245. SO WE SHOULD ONLY MERGE THIS PR IF ALSO #245 IS READY
+  /*
+   * calculate derived quantities such as vorticity, divergence, etc.
+   */
+  calculate_additional_vectors(velocity, time, time_step_number);
```

Comment:
I pulled out the calculation of the additional vectors such that the interface of evaluate() can be as in all other cases. This is not a good idea without https://github.com/exadg/exadg/pull/245 since now additional vectors are computed every time-step. I think we should keep it this way and merge this PR and https://github.com/exadg/exadg/pull/245 at the same time, once https://github.com/exadg/exadg/pull/245 is ready.

Reply:
I think the computations are only invoked if the output module is activated. So I think this is acceptable as an intermediate solution.